File: pythonStuff/leetcode53.py
#This solution is called Kadane's algorithm
#Runs in O(n) time

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        pointer = 1
        while pointer <= len(nums)-1:
            if nums[pointer-1] > 0:
                nums[pointer] += nums[pointer-1]
            pointer += 1
        return max(nums)

File: javaStuff/leetcode53.java
class Solution {
    public int maxSubArray(int[] nums) {
        int currentMax = nums[0], maxEnd = nums[0];
        for(int i = 1; i < nums.length; i++){
            maxEnd = Math.max(maxEnd + nums[i], nums[i]);
            currentMax = Math.max(currentMax, maxEnd);
        }
        return currentMax;
    }
}